plugins/org.eclipse.n4js/src/org/eclipse/n4js/resource/N4JSResource.java

```diff
@@ -259,6 +265,32 @@ public N4JSResource() {
 	}
 
 	/**
+	 * Returns the {@link ASTMetaInfoCache} (in states {@link #isFullyProcessed() "Fully Processed"} and during the
+	 * transition from "Fully Initialized" to "Fully Processed", i.e. during post-processing) or throws an exception if
+	 * the cache is unavailable (in all other states).
+	 */
+	public ASTMetaInfoCache getASTMetaInfoCache() {
+		if (astMetaInfoCache == null) {
+			if (!isFullyProcessed() && !isPostProcessing()) {
```

Comment:
The inner if check `if (!isFullyProcessed() && !isPostProcessing())`  could be placed at the beginning of the method?

Reply:
This check is used only to adjust the error message of the exception thrown immediately afterwards. If `astMetaInfoCache` is non-null, I think it is better to just return it.